impeller/golden_tests/metal_screenshot.mm

```diff
@@ -12,10 +12,7 @@
   pixel_data_ = CGDataProviderCopyData(data_provider);
 }
 
-MetalScreenshot::~MetalScreenshot() {
-  CFRelease(pixel_data_);
-  CGImageRelease(cg_image_);
```

Comment:
> If preferred, it'd be pretty straightforward to update CFRef to a templated type

Wouldn't we just need to add a template specialization for the existing class for any type with its own release function?

It would be nice to do that for `CGImageRef` from a technical correctness standpoint, but I'm also not going to lose any sleep over the idea that Apple might one day make the specialized releasers *not* just null-safe versions of `CFRelease`, in violation of both the official documentation and decades of established behavior.